test/e2e/storage/testsuites/subpath.go

```diff
@@ -580,13 +580,17 @@ func testPodFailSubpath(f *framework.Framework, pod *v1.Pod) {
 }
 
 func testPodFailSubpathError(f *framework.Framework, pod *v1.Pod, errorMsg string) {
-	By(fmt.Sprintf("Creating pod %s", pod.Name))
+	By(fmt.Sprintf("Creating pod %s (expected to fail with subpath error)", pod.Name))
 	pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
 	Expect(err).ToNot(HaveOccurred(), "while creating pod")
 	defer func() {
 		framework.DeletePodWithWait(f, f.ClientSet, pod)
 	}()
-	err = framework.WaitForPodRunningInNamespace(f.ClientSet, pod)
+
+	// Check if the pod starts running - that would be unexpected
+	// We will delay by the full timeout, so keep it reasonably short: 30s
+	// This is primarily a sanity check; the actual check is the error event.
+	err = framework.WaitTimeoutForPodRunningInNamespace(f.ClientSet, pod.Name, pod.Namespace, 30*time.Second)
```

Comment:
there's already a PodStartShortTimeout const set to 2 mins in framework/util.go.  Do we want to look into shortening that across the board (all e2es) or define another short parameter just for this test?

Reply:
So I did debate just removing this "wait for pod running" check entirely - the only purpose it's serving is to provide a clearer diagnostic if the event is not posted (did the pod start?)  The problem is that because we don't expect the pod to start, whatever we wait for will take the whole timeout time - and this test is run repeatedly - at first glance it looked like it was adding ~2 hours to the serial test @ 5 minutes timeout.

I figured 30 seconds was sort of a compromise between "if the pod is starting, occasionally we'll catch it" and the delay becoming unreasonable.

I guess an alternative approach could be to remove WaitTimeoutForPodRunningInNamespace entirely, and wait for the event; once we get the event we then check the pod is not running, but on error we still log the state of the pod.  Though maybe the normal diagnostic dump after a failed test would be sufficient and there's no point special-case logging the pod (?).

Reply:
I think the second approach works too and could return even earlier than the 30 second timeout.

Reply:
Cool - you mean just removing the "WaitTimeoutForPodRunningInNamespace" entirely?   Do you think we should try to print / check the pod status?

Reply:
This change seems unnecessary today because of https://github.com/kubernetes/kubernetes/commit/790de995f35c53b79d0f5d7bf65e3bf1d0c5a197